### PR TITLE
Migrate docker containers to latest tags

### PR DESCRIPTION
**Pull request by michellemcdaniel**
This change moves all of the docker images to the latest tag as part of https://github.com/dotnet/dnceng/issues/1294.

This change also moves the containers that no longer exist to replacements (for example, centos-8-rpmpkg no longer exists, and that was moved to centos-7-rpmpkg).

### CODE REVIEW

**Comment by adiaaida** [comment]
@wfurt @wtgodbe I think the change to how fpm is installed on the centos-7-rpmpkg container is causing the "FPM cannot be found" issue (https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/78d25937ee7c831b1c368ff0b0c3dce043cb98cb/src/centos/7/rpmpkg/Dockerfile#L11). On the centos-8-rpmpkg container (which no longer exists), we just ran 

```
gem install --no-document fpm 
```

Do you know if the current way is installing fpm to a different place other than /usr/bin/env?

**Comment by wtgodbe** [comment]
I do not, I don't know much about how that works - I just copy/pasted the original workaround from aspnetcore 😢

**Comment by adiaaida** [comment]
I'll create a docker container tomorrow and try to figure out where fpm is installed on it

**Comment by wfurt** [comment]
`fpm` is only needed to create msquic packages. Where so you see the error? 
Also the quic/http3 failures looks suspicious.  That was fixed by #76267 yesterday.
The images need at least msquic 2.1.2.

**Comment by wfurt** [comment]
cc: @rzikm

**Comment by adiaaida** [comment]
> fpm is only needed to create msquic packages. Where so you see the error?
> Also the quic/http3 failures looks suspicious. That was fixed by https://github.com/dotnet/runtime/pull/76267 yesterday.
The images need at least msquic 2.1.2.

The errors are in a bunch of the legs, for example: /_build/results?buildId=40304&view=logs&j=36f5df04-f069-50e9-49af-755a9549bd25&t=bdd8d2a1-63f8-59f1-8e0a-89cd06102b80&l=48

The msquic changes will roll out to production tomorrow, so I can rerun those legs post-rollout.

**Comment by wfurt** [comment]
ah, I was actually wrong @adiaaida. I add use of `fpm` recently for msquic, but that was mimicked after some runtime code. 
It seems like the executable is truly missing in the docker image. We will probably need to investigate and rebuild the container(s)

**Comment by adiaaida** [comment]
I think I have a working version of the docker image, after much hair pulling. I would like to build runtime/this scenario inside the container to make sure it works properly. Who is the right person to help me with the correct set of build commands?

**Comment by wfurt** [comment]
we may link it as well if needed. I don't think each stage/pipeline shows the commands at the beginning and they differ e.g. I don't think there is simple command set....

**Comment by michellemcdaniel** [comment]
@dotnet/runtime-infrastructure I believe the failures in extra-platforms are all known failures. The base runtime pipeline is green. PTAL.

**Comment by wfurt** [comment]
Only one suspicious is AOT build failure
```
  Optimizing assemblies for size may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink
  Generating native code
EXEC : error : Failed to load assembly 'SQLitePCLRaw.core' [/__w/1/s/src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/ComInterfaceGenerator.Unit.Tests.csproj]
##[error]EXEC(0,0): error : (NETCORE_ENGINEERING_TELEMETRY=Build) Failed to load assembly 'SQLitePCLRaw.core'
```
Do you know @jkotas if that is normal/expected?

**Comment by michellemcdaniel** [comment]
The AOT failures are https://github.com/dotnet/runtime/issues/77476. I believe there is a PR to fix it or disable a test, but it is unrelated to this change

**Comment by jkotas** [comment]
> Only one suspicious is AOT build failure

This should be fixed by https://github.com/dotnet/runtime/pull/77522 that was just merged.

**Comment by michellemcdaniel** [comment]
I might need to merge in main to get those changes

**Comment by michellemcdaniel** [comment]
Alrighty, I think this is good. I believe all of the failures currently in extra-platforms are on native runs, not on docker containers. PTAL

**Comment by michellemcdaniel** [comment]
@dotnet/runtime-infrastructure please take a look

File: eng/pipelines/common/templates/pipeline-with-resources.yml [hunk 1]
```diff
@@ -5,75 +5,75 @@ parameters:
...
       env:
         ROOTFS_DIR: /crossrootfs/armel
 
     - container: debpkg
-      image: mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-debpkg-20220504035737-cfdd435
+      image: mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-debpkg
 
     - container: rpmpkg
-      image: mcr.microsoft.com/dotnet-buildtools/prereqs:centos-8-rpmpkg-20210714125410-daa5116
+      image: mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7-rpmpkg
```
Comment (agocke) [inline eng/pipelines/common/templates/pipeline-with-resources.yml:77]: I think we upgraded this due to the Ruby scripts that we run to make packaging not being available on centos 7. Moving backwards might break the official build.

**Review by agocke — APPROVED** [review]

File: eng/pipelines/common/templates/pipeline-with-resources.yml (hunk 1 above)
Comment (wfurt) [reply eng/pipelines/common/templates/pipeline-with-resources.yml:77]: good catch. I'm not sure if this was intentional @michellemcdaniel ?

Comment (michellemcdaniel) [reply eng/pipelines/common/templates/pipeline-with-resources.yml:77]: I fixed this in https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/718. Ruby is available on the centos 7 containers. Centos 8 is eol and we no longer have containers for it.

Comment (wfurt) [reply eng/pipelines/common/templates/pipeline-with-resources.yml:77]: right. I forget about the 8 story.

**Pull request merged**